src/ur5e_arm.cpp

```diff
@@ -403,6 +403,54 @@ void UR5eArm::send_trajectory(const std::vector<vector6d_t>& p_p,
 
 // helper function to read a data packet and send a noop message
 void UR5eArm::read_and_noop() {
+    // check to see if an estop has occurred.
+    std::string status;
```

Comment:
Ok a lot more is happening in this function now.  Can we break this up into two parts?  One for managing EStops and restarting communication in fail cases and one for actually reading and nooping?  I'm open to a bunch of ideas on how to organize this tbh

Reply:
personally I liked having all of the code live in the same function.

if peeps want me to have `read_joint_keep_alive()` call a `handle_estop` and `handle_packet`(maybe `handle_noop`?) then I'm okay with that

Reply:
I'm ok with keeping this in one function for now.